File: .kotlin_alpha/services/sns/src/main/kotlin/com/kotlin/sns/PublishTopic.kt
//snippet-sourcedescription:[PublishTopic.kt demonstrates how to publish an Amazon Simple Notification Service (Amazon SNS) topic.]
//snippet-keyword:[AWS SDK for Kotlin]
//snippet-keyword:[Code Sample]
//snippet-keyword:[Amazon Simple Notification Service]
//snippet-sourcetype:[full-example]
//snippet-sourcedate:[11/05/2021]
//snippet-sourceauthor:[scmacdon- AWS]

/*
   Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
   SPDX-License-Identifier: Apache-2.0
*/

package com.kotlin.sns

//snippet-start:[sns.kotlin.PublishTopic.import]
import aws.sdk.kotlin.services.sns.SnsClient
import aws.sdk.kotlin.services.sns.model.PublishRequest
import kotlin.system.exitProcess
//snippet-end:[sns.kotlin.PublishTopic.import]

suspend fun main(args:Array<String>) {

    val usage = """
          Usage: 
            <message> <topicArn>

        Where:
            message - the message text to send.
            topicArn - the ARN of the topic to publish.
            """

    if (args.size != 2) {
        println(usage)
        exitProcess(0)
    }

    val message = args[0]
    val topicArn = args[1]
    pubTopic(topicArn, message)
}

//snippet-start:[sns.kotlin.PublishTopic.main]
suspend fun pubTopic(topicArnVal: String, messageVal: String) {

    val request = PublishRequest{
        message = messageVal
        topicArn = topicArnVal
     }

    SnsClient { region = "us-east-1" }.use { snsClient ->
        val result = snsClient.publish(request)
        println("${result.messageId} message sent.")
    }
}
//snippet-end:[sns.kotlin.PublishTopic.main]

File: php/example_code/sns/PublishTopic.php
<?php
// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
// SPDX-License-Identifier: Apache-2.0

/*
 *  ABOUT THIS PHP SAMPLE: This sample is part of the SDK for PHP Developer Guide topic
 *
 *
 */
// snippet-start:[sns.php.publish_topic.complete]
// snippet-start:[sns.php.publish_topic.import]
require 'vendor/autoload.php';

use Aws\Exception\AwsException;
use Aws\Sns\SnsClient;

// snippet-end:[sns.php.publish_topic.import]

/**
 * Sends a message to an Amazon SNS topic.
 *
 * This code expects that you have AWS credentials set up per:
 * https://docs.aws.amazon.com/sdk-for-php/v3/developer-guide/guide_credentials.html
 */

// snippet-start:[sns.php.publish_topic.main]
$SnSclient = new SnsClient([
    'profile' => 'default',
    'region' => 'us-east-1',
    'version' => '2010-03-31'
]);

$message = 'This message is sent from a Amazon SNS code sample.';
$topic = 'arn:aws:sns:us-east-1:111122223333:MyTopic';

try {
    $result = $SnSclient->publish([
        'Message' => $message,
        'TopicArn' => $topic,
    ]);
    var_dump($result);
} catch (AwsException $e) {
    // output error message if fails
    error_log($e->getMessage());
}

// snippet-end:[sns.php.publish_topic.main]
// snippet-end:[sns.php.publish_topic.complete]
